app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/gainorlossafterreliefs/GainOrLossAfterReliefsController.scala

```diff
@@ -87,18 +98,25 @@ class GainOrLossAfterReliefsController @Inject() (
           .bindFromRequest()
           .fold(
             formWithErrors =>
-              BadRequest(
-                gainOrLossAfterReliefsPage(
-                  formWithErrors,
-                  answer.fold(controllers.returns.routes.TaskListController.taskList())(_ =>
-                    routes.GainOrLossAfterReliefsController.checkYourAnswers()
-                  ),
-                  fillingOutReturn.subscribedDetails.isATrust,
-                  draftReturn.representativeType(),
-                  draftReturn.triageAnswers().isLeft,
-                  fillingOutReturn.isAmendReturn
-                )
-              ),
+              glarCalculatorEligibilityUtil
+                .isEligibleForFurtherReturnOrAmendCalculation(fillingOutReturn)
+                .fold(
+                  _ => InternalServerError,
+                  includeBreakdown =>
+                    BadRequest(
+                      gainOrLossAfterReliefsPage(
+                        formWithErrors,
+                        answer.fold(controllers.returns.routes.TaskListController.taskList())(_ =>
+                          routes.GainOrLossAfterReliefsController.checkYourAnswers()
+                        ),
+                        fillingOutReturn.subscribedDetails.isATrust,
+                        draftReturn.representativeType(),
+                        draftReturn.triageAnswers().isLeft,
+                        fillingOutReturn.isAmendReturn,
+                        if (includeBreakdown) buildGlarBreakdown(draftReturn) else None
```

Comment:
if you can't get a breakdown at this point in the journey that's an error case - you could do something like:
```scala
if(includeBreakdown)  buildBreakdown.fold(Redirect(taskListPage))(...)
else ...
```
and maybe factor out the `BadRequest(gainOrLossAfterReliefsPage(...))` part out somehow


```